gobblin-service/src/main/java/gobblin/service/modules/monitoring/JobStatusRetriever.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gobblin.service.modules.monitoring;
+
+import java.util.Iterator;
+
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+
+import gobblin.annotation.Alpha;
+
+
+/**
+ * Retriever for {@link JobStatus}.
+ */
+@Alpha
+public abstract class JobStatusRetriever {
+
+  public abstract Iterator<JobStatus> getJobStatusesByFlowNameAndFlowGroup(String flowName, String flowGroup);
+
+  /**
+   * Get the latest {@link JobStatus}es that belongs to the same latest flow execution. Currently, latest flow execution
+   * is decided by comparing {@link JobStatus#getFlowId()}.
+   */
+  public Iterator<JobStatus> getLatestJobStatusByFlowNameAndGroup(String flowName, String flowGroup) {
+    Iterator<JobStatus> jobStatusIterator = getJobStatusesByFlowNameAndFlowGroup(flowName, flowGroup);
+    Multimap<String, JobStatus> flowIdToJobStatusMultiMap = ArrayListMultimap.create();
+    String latestFlowId = null;
+    while (jobStatusIterator.hasNext()) {
+      JobStatus jobStatus = jobStatusIterator.next();
+      flowIdToJobStatusMultiMap.put(jobStatus.getFlowId(), jobStatus);
+      if (latestFlowId == null || jobStatus.getFlowId().compareTo(latestFlowId) > 0) {
+        latestFlowId = jobStatus.getFlowId();
+      }
+    }
+    return flowIdToJobStatusMultiMap.get(latestFlowId).iterator();
```

Comment:
Can we push the filtering down to the db by querying first for the highest flow id for the flow then getting all job statuses for that flow id?